packages/kbn-scout-reporting/src/reporting/report/persistence/index.ts

```diff
@@ -0,0 +1,141 @@
+import * as indexTemplates from './index_templates';
+
+export const dataStreamName = 'scout-test-events-kibana';
+
+export class ScoutReportDataStream {
+  private log: ToolingLog;
+
+  constructor(private es: ESClient, log?: ToolingLog) {
+    this.log = log || new ToolingLog();
+  }
+
+  async exists() {
+    return await this.es.indices.exists({ index: dataStreamName });
+  }
+
+  async initialize() {
+    await this.setupComponentTemplates();
+    await this.setupIndexTemplate();
+
+    if (await this.exists()) {
+      return;
+    }
+
+    this.log.info(`Creating data stream '${dataStreamName}'`);
+    await this.es.indices.createDataStream({
+      name: dataStreamName,
+    });
+  }
+
+  async setupComponentTemplates() {
+    for (const template of [
+      componentTemplates.buildkiteMappings,
+      componentTemplates.reporterMappings,
+      componentTemplates.testRunMappings,
+      componentTemplates.suiteMappings,
+      componentTemplates.testMappings,
+    ]) {
+      const templateExists = await this.es.cluster.existsComponentTemplate({ name: template.name });
+      if (!templateExists) {
+        this.log.info(`Creating component template '${template.name}'`);
+        await this.es.cluster.putComponentTemplate(template);
+        continue;
+      }
+
+      // Template exists but might need to be updated
+      const newTemplateVersion = template.version || 0;
+      const existingTemplateVersion =
+        (await this.es.cluster.getComponentTemplate({ name: template.name })).component_templates[0]
+          .component_template.version || 0;
+
+      if (existingTemplateVersion >= newTemplateVersion) {
+        this.log.info(`Component template '${template.name} exists and is up to date.`);
+        continue;
+      }
+
+      this.log.info(
+        `Updating component template '${template.name}' (version ${existingTemplateVersion} -> ${newTemplateVersion})`
+      );
+      await this.es.cluster.putComponentTemplate(template);
+    }
+  }
+
+  async setupIndexTemplate() {
+    const template = indexTemplates.testEvents;
+    const templateExists: boolean = await this.es.indices.existsIndexTemplate({
+      name: template.name,
+    });
+
+    if (!templateExists) {
+      this.log.info(`Creating index template '${template.name}'`);
+      await this.es.indices.putIndexTemplate(template);
+      return;
+    }
+
+    // Template exists but might need to be updated
+    const newTemplateVersion = template.version || 0;
+    const existingTemplateVersion =
+      (await this.es.indices.getIndexTemplate({ name: template.name })).index_templates[0]
+        .index_template.version || 0;
+
+    if (existingTemplateVersion >= newTemplateVersion) {
+      this.log.info(`Index template '${template.name} exists and is up to date.`);
+      return;
+    }
+
+    this.log.info(
+      `Updating index template '${template.name}' (version ${existingTemplateVersion} -> ${newTemplateVersion})`
+    );
+  }
```

Comment:
Looks like the actual template update call is missing? I'd expect something like `await this.es.indices.putIndexTemplate(template);` here.

Reply:
Eagle-eyed as always. You're right. 👍🏽